### PR TITLE
Multiple deploy_website.sh fixes

### PR DESCRIPTION
**Pull request by Egorand**
- Make OSX sed command the default
- Undo the results of sed to jump branches
- --set-upstream before pushing to gh-pages

### CODE REVIEW

File: deploy_website.sh [hunk 1]
```diff
@@ -40,9 +40,13 @@ cp CONTRIBUTING.md docs/contributing.md
 # Build the site and push the new files up to GitHub
 mkdocs gh-deploy
 
+# Undo any file changes to be able to jump branches.
+git checkout -- .
```
Comment (jrodbx) [inline deploy_website.sh]: would `git reset --soft` suffice?

File: deploy_website.sh [hunk 2]
```diff
@@ -40,9 +40,13 @@ cp CONTRIBUTING.md docs/contributing.md
...
+
 # Restore Javadocs from 2.x
 git checkout gh-pages
 git cherry-pick 0e1192aaa1d074c9703748fae100daef707218d4
+git push --set-upstream origin gh-pages
 git push
```
Comment (jrodbx) [inline deploy_website.sh]: combine these two lines?

File: deploy_website.sh (hunk 1 above)
Comment (Egorand) [reply deploy_website.sh]: No, gotta be `--hard`. What's the difference anyway?

File: deploy_website.sh (hunk 2 above)
Comment (pavlospt) [reply deploy_website.sh]: There is also a shorter version with `git push -u origin gh-pages` if you want!

File: deploy_website.sh (hunk 1 above)
Comment (pavlospt) [reply deploy_website.sh]: `reset --soft` returns to the desired hash while retaining the changes from all the commits between the `HEAD` and desired hash.

Comment (Egorand) [reply deploy_website.sh]: I'm trying to undo the changes introduced by the previous commands in this script, and `git reset --soft` doesn't seem to cut it. `git checkout -- .` works, so I'll keep it.

Comment (pavlospt) [reply deploy_website.sh]: Yes, you need a hard reset here, or a stash!

**Pull request merged**